.github/workflows/create-tag.yml

```diff
@@ -7,9 +7,11 @@ on:
   push:
     branches:
       - main
+    paths:
+      - package.json
 
 permissions:
-  contents: write
+  contents: read
```

Comment:
The GH API to create a release (https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release) says it needs:

> The token must have at least one of the following permission sets:
> 
>     contents:write
>     contents:write and workflows:write

I guess this is bypassing using the GH token for this job and instead is using the `CREATE_TAG_TOKEN` secret that is manually maintained for this repo?

Reply:
> is using the CREATE_TAG_TOKEN secret that is manually maintained for this repo?

Correct. Unfortunately, we cannot use the ephemeral GH token if GitHub events are required.

In this particular the GitHub event is needed, to be able to run the [release.yml workflow](https://github.com/elastic/opbeans-node/blob/7acf953174dd470dae1919324ecd4b231be8a9fc/.github/workflows/release.yml#L6).


GitHub does not support GitHub events when the actor is the GitHub bot, a way to prevent recursivity

Reply:
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow

> When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN, with the exception of workflow_dispatch and repository_dispatch, will not create a new workflow run